### PR TITLE
ARIES-1746: Add SPIFly support for InitialContextFactory

### PR DESCRIPTION
**Pull request by thusithathilina**
According to the OSGi spec when we register an InitialContextFactory, we need to register that for both implementation and InitialContextFactory classes. This PR will provide that support.

### CODE REVIEW

**Comment by thusithathilina** [comment]
It would be great if someone can look into this if possible. We would love to have a release with this.

File: spi-fly/spi-fly-core/src/main/java/org/apache/aries/spifly/ProviderBundleTrackerCustomizer.java [hunk 1]
```diff
@@ -169,6 +171,14 @@ public ProviderBundleTrackerCustomizer(BaseActivator activator, Bundle spiBundle
...
                             properties = findServiceRegistrationProperties(bundle.getHeaders(), registrationClassName, className);
 
+                        String[] registrationClassNames;
+                        if (registrationClassName.equals("javax.naming.spi.InitialContextFactory")) {
+                            registrationClassNames = new String[2];
+                            registrationClassNames[1] = cls.getName();
+                        } else {
+                            registrationClassNames = new String[1];
+                        }
+                        registrationClassNames[0] = registrationClassName;
```
Comment (bosschaert) [inline spi-fly/spi-fly-core/src/main/java/org/apache/aries/spifly/ProviderBundleTrackerCustomizer.java:181]: This change is very specific to the javax.naming.spi.InitialContextFactory and does not belong in SPI-Fly IMHO.

**Comment by bosschaert** [comment]
I left a comment on the issue in Apache JIRA on how you could get your own bundle to do something similar. SPI Fly should **not** have specific logic for just `javax.naming.spi.InitialContextFactory`

See: https://issues.apache.org/jira/browse/ARIES-1746

**Comment by cschneider** [comment]
We have now moved the aries sources to git. Unfortunately this means that there are now conflicts with the existing pull request. Can you rebase your changes to the current master?

**Comment by cschneider** [comment]
Closing for inactivity. Please create a new PR with the changes based on current trunk.

**Pull request closed without merging**